flaml/autogen/agentchat/groupchat.py

```diff
@@ -0,0 +1,204 @@
+import sys
+from typing import Dict, List, Optional, Union
+import random
+from .agent import Agent
+from .responsive_agent import ResponsiveAgent
+
+
+class Room(ResponsiveAgent):
+    """(WIP) A room agent that can broadcast messages to multiple agents."""
+
+    agents: List[ResponsiveAgent]
+
+    def __init__(
+        self,
+        agents: Optional[List[ResponsiveAgent]] = None,
+        name: Optional[str] = "room",
+        # unlimited consecutive auto reply by default
+        max_consecutive_auto_reply: Optional[int] = sys.maxsize,
+        human_input_mode: Optional[str] = "NEVER",
+        **kwargs,
+    ):
+        super().__init__(
+            name=name,
+            max_consecutive_auto_reply=max_consecutive_auto_reply,
+            human_input_mode=human_input_mode,
+            **kwargs,
+        )
+        self.register_class_specific_reply(Agent, self._generate_reply_for_agent)
+        self.agents = agents or []
+
+    def _generate_reply_for_agent(
+        self,
+        messages: Optional[List[Dict]] = None,
+        default_reply: Optional[Union[str, Dict]] = None,
+        sender: Optional[Agent] = None,
+    ) -> Union[str, Dict, None]:
+        """Broadcast a message from an agent in the group chat."""
+        if messages is None:
+            messages = self._oai_messages[sender.name]
+        message = messages[-1]
+        # set the name to sender's name if the role is not function
+        if message["role"] != "function":
+            message["name"] = sender.name
+        # broadcast the message to all agents except the sender
+        for agent in self.agents:
+            if agent != sender:
+                self.send(message, agent)
+
+
+class ChatManagerAgent(ResponsiveAgent):
```

Comment:
Maybe we can move
- `select_next_speaker`
- `participant_information`

into `ChatManagerAgent`?

The `Room` will have
- A `ChatManagerAgent`
- A group of `ResponsiveAgent` as participants

In each round, `ChatManagerAgent` will be responsible for selecting the next speaker from the participants. `ChatManagerAgent` will also be responsible for receiving the message from next speaker.

Therefore, No extra change would be required for `ResponsiveAgent` to be a participant in a group chat.